app/src/test/java/com/example/habitsmasher/ui/dashboard/HabitValidatorTest.java

```diff
@@ -37,45 +44,67 @@ public void setUp() {
-        assertFalse(_validator.isHabitValid(GOOD_TITLE, "", TODAY, GOOD_TRACKER, ));
+        assertFalse(_validator.isHabitValid(GOOD_TITLE, "", TODAY, GOOD_TRACKER, GOOD_PRIVACY_BUTTONS));
     }
 
     @Test
     public void isHabitValid_emptyDate_expectHabitIsInvalid() {
-        assertFalse(_validator.isHabitValid(GOOD_TITLE, GOOD_REASON, "", GOOD_TRACKER, ));
+        assertFalse(_validator.isHabitValid(GOOD_TITLE, GOOD_REASON, "", GOOD_TRACKER, GOOD_PRIVACY_BUTTONS));
     }
     @Test
     public void isHabitValid_emptyDays_expectHabitIsInvalid() {
-        assertFalse(_validator.isHabitValid(GOOD_TITLE, GOOD_REASON, TODAY, BAD_TRACKER, ));
+        assertFalse(_validator.isHabitValid(GOOD_TITLE, GOOD_REASON, TODAY, BAD_TRACKER, GOOD_PRIVACY_BUTTONS));
     }
 
     @Test
     public void isHabitValid_titleTooLong_expectHabitIsInvalid() {
-        assertFalse(_validator.isHabitValid(LONG_TITLE, GOOD_REASON, TODAY, GOOD_TRACKER, ));
+        assertFalse(_validator.isHabitValid(LONG_TITLE, GOOD_REASON, TODAY, GOOD_TRACKER, GOOD_PRIVACY_BUTTONS));
     }
 
     @Test
     public void isHabitValid_reasonTooLong_expectHabitIsInvalid() {
-        assertFalse(_validator.isHabitValid(GOOD_TITLE, LONG_REASON, TODAY, GOOD_TRACKER, ));
+        assertFalse(_validator.isHabitValid(GOOD_TITLE, LONG_REASON, TODAY, GOOD_TRACKER, GOOD_PRIVACY_BUTTONS));
     }
 
     @Test
     public void isHabitValid_titleExactly20Characters_expectHabitIsValid() {
-        assertTrue(_validator.isHabitValid(TITLE_20_CHARACTERS, GOOD_REASON, TODAY, GOOD_TRACKER, ));
+        assertTrue(_validator.isHabitValid(TITLE_20_CHARACTERS, GOOD_REASON, TODAY, GOOD_TRACKER, GOOD_PRIVACY_BUTTONS));
     }
 
     @Test
     public void isHabitValid_reasonExactly30Characters_expectHabitIsValid() {
-        assertTrue(_validator.isHabitValid(GOOD_TITLE, REASON_30_CHARACTERS, TODAY, GOOD_TRACKER, ));
+        assertTrue(_validator.isHabitValid(GOOD_TITLE, REASON_30_CHARACTERS, TODAY, GOOD_TRACKER, GOOD_PRIVACY_BUTTONS));
+    }
+
+    @Test
+    public void isHabitValid_noPrivacyButtonsSelected_expectHabitIsInvalid(){
+        assertFalse(_validator.isHabitValid(GOOD_TITLE, GOOD_REASON, TODAY, GOOD_TRACKER, BAD_PRIVACY_BUTTONS));
+    }
+
+    @Test
+    public void isHabitValid_bothPrivacyButtonsSelected_expectHabitIsInvalid(){
+        BAD_PRIVACY_BUTTONS.setButtons(true, true);
+        assertFalse(_validator.isHabitValid(GOOD_TITLE, GOOD_REASON, TODAY, GOOD_TRACKER, BAD_PRIVACY_BUTTONS));
+    }
+
+    @Test
+    public void isHabitValid_publicButtonSelected_expectHabitIsValid(){
+        assertTrue(_validator.isHabitValid(GOOD_TITLE, GOOD_REASON, TODAY, GOOD_TRACKER, GOOD_PRIVACY_BUTTONS));
+    }
+
+    @Test
+    public void isHabitValid_privateButtonSelected_expectHabitIsValid(){
+        GOOD_PRIVACY_BUTTONS.setButtons(false, true);
+        assertTrue(_validator.isHabitValid(GOOD_TITLE, GOOD_REASON, TODAY, GOOD_TRACKER, GOOD_PRIVACY_BUTTONS));
```

Comment:
Made more unit tests after fixing them. Can't believe I forgot to make the tests too...